tests/subcommands.rs

```diff
@@ -232,11 +219,43 @@ fn subcmd_did_you_mean_output_ambiguous() {
 }
 
 #[test]
-#[cfg(feature = "suggestions")]
+#[cfg(feature="suggestions")]
 fn subcmd_did_you_mean_output_arg() {
-    let app =
-        App::new("dym").subcommand(App::new("subcmd").arg("-s --subcmdarg [subcmdarg] 'tests'"));
-    assert!(utils::compare_output(app, "dym --subcm foo", DYM_ARG, true));
+        static EXPECTED: &'static str = "error: Found argument '--subcmarg' which wasn't expected, or isn't valid in this context
+
+\tDid you mean to put '--subcmdarg' after the subcommand 'subcmd'?
+
+If you tried to supply `--subcmarg` as a PATTERN use `-- --subcmarg`
+
+USAGE:
+    dym [SUBCOMMAND]
+
+For more information try --help";
+
+    let app = App::new("dym")
+        .subcommand(App::new("subcmd")
+            .arg("-s --subcmdarg [subcmdarg] 'tests'") );
+    assert!(utils::compare_output(app, "dym --subcmarg subcmd", EXPECTED, true));
+}
+
+#[test]
+#[cfg(feature="suggestions")]
+fn subcmd_did_you_mean_output_arg_false_positives() {
+    static EXPECTED: &'static str = "error: Found argument '--subcmarg' which wasn't expected, or isn't valid in this context
+
+\tDid you mean to put '--subcmdarg' after the subcommand 'subcmd'?
```

Comment:
```suggestion
```

Reply:
I'm not sure about that change, it fails to pass test subcmd_did_you_mean_output_arg_false_positives then

Reply:
You just copied the files from that commit. It won't work like since we changed the files and where the logic happens also. You need to take the logic from that commit and understand it and find out where to add it in the master branch.

As I said, the tests are now correct. You just haven't fixed the bug by updating the logic correctly.